hack/release/process-changelog.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Run this script with "go run process-changelog.go <PATH to CHANGELOG> > <PATH to new CHANGELOG>".
+// The script will:
+//  * Add links to PRs, by replacing all instances of [#<NUM>] with [#<NUM>](<link to PR NUM>)
+//  * Add links to the Github profiles of PR authors. It will look for instances of [@<AUTHOR>] in
+//    the CHANGELOG and generate the appropriate Markdown links at the end of the file.
```

Comment:
Do you think it would be helpful to even generate the content of the changelog. Like in K8s, each PR has a section about release note. Then maybe the script can gather all PRs since last minor release (if this is for minor release) or patch release (if this is for patch release) and extract the "release note" sections?

Reply:
I think it would be helpful, but so far it's been manageable to generate the CHANGELOG just before the release because the volume is not crazy. I want to avoid making the PR process more complicated for contributors for now, and I think it's been useful to group PRs by feature in some cases to make the CHANGELOG more organized. But if the project grows, I think we would have to go with a solution like the K8s one.

Reply:
Sounds good to me.